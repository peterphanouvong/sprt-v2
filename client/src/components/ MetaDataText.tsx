import { Text } from "@chakra-ui/react";
import React from "react";

interface Props {
  display?: string;
}

const MetaDataText: React.FC<Props> = ({ children, display = "inline" }) => {
  return (
    <Text display={display} color="GrayText" fontSize="sm">
      {children}
    </Text>
  );
};

export { MetaDataText };
