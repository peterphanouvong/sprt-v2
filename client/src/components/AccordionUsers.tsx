import { CheckIcon } from "@chakra-ui/icons";
import {
  Box,
  Accordion,
  AccordionItem,
  AccordionButton,
  AccordionIcon,
  AccordionPanel,
  ListItem,
  Text,
} from "@chakra-ui/react";
import React from "react";
import { User } from "../generated/graphql";

interface Props {
  userType: string;
  userList: User[];
  numHighlighted?: number;
}

const AccordionUsers: React.FC<Props> = ({
  userType,
  userList,
  numHighlighted = 0,
}) => {
  return (
    <Box mt={4}>
      <Accordion allowToggle>
        <AccordionItem>
          <AccordionButton
            _hover={{
              backgroundColor: "gray.50",
            }}
          >
            <Box flex='1' textAlign='left'>
              <Text fontWeight='medium'>{userType}</Text>
            </Box>
            <AccordionIcon />
          </AccordionButton>
          <AccordionPanel p={0}>
            {userList.map((attendee, index) => {
              const isHighlighted = index < numHighlighted;
              return (
                <ListItem
                  paddingY={1}
                  paddingX={4}
                  listStyleType='none'
                  key={attendee.id}
                  background={isHighlighted ? "green.50" : "yellow.50"}
                  _hover={{
                    backgroundColor: isHighlighted ? "green.100" : "yellow.100",
                  }}
                >
                  <Box
                    fontSize='sm'
                    color={isHighlighted ? "green.900" : "yellow.900"}
                    display='flex'
                    justifyContent='space-between'
                    alignItems='center'
                  >
                    {attendee.username}
                    {isHighlighted ? (
                      <CheckIcon color='green' />
                    ) : (
                      <Text>(waitlist)</Text>
                    )}
                  </Box>
                </ListItem>
              );
            })}
          </AccordionPanel>
        </AccordionItem>
      </Accordion>
    </Box>
  );
};

export { AccordionUsers };
