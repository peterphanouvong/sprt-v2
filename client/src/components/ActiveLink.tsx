import React from "react";
import NextLink from "next/link";
import { Box } from "@chakra-ui/layout";
import { useRouter } from "next/router";

interface Props {
  href: string;
}

const ActiveLink: React.FC<Props> = ({ children, href }) => {
  const router = useRouter();
  const isActive = router.pathname === href;
  return (
    <Box
      _hover={{ borderBottom: "1px solid", borderColor: "gray.300" }}
      borderBottom={isActive ? "1px solid" : ""}
      borderColor={isActive ? "gray.300" : ""}
    >
      <NextLink href={href}>
        <a>{children}</a>
      </NextLink>
    </Box>
  );
};

export { ActiveLink };
