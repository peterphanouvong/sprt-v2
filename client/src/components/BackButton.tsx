import { ChevronLeftIcon } from "@chakra-ui/icons";
import { ButtonProps, IconButton } from "@chakra-ui/react";
import { useRouter } from "next/router";
import React from "react";

type Props = ButtonProps;

const BackButton: React.FC<Props> = ({ ...props }) => {
  const router = useRouter();
  return (
    <IconButton
      isRound={true}
      aria-label="back-button"
      icon={<ChevronLeftIcon />}
      onClick={() => router.back()}
      {...props}
    />
  );
};

export { BackButton };
