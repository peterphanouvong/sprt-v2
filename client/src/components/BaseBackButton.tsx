import { ChevronLeftIcon } from "@chakra-ui/icons";
import { ButtonProps, IconButton } from "@chakra-ui/react";
import { useRouter } from "next/router";
import React from "react";

type Props = ButtonProps;

const BaseBackButton: React.FC<Props> = ({ ...props }) => {
  const router = useRouter();
  return (
    <IconButton
      isRound={true}
      colorScheme="gray"
      variant="solid"
      aria-label="back-button"
      icon={<ChevronLeftIcon />}
      onClick={() => router.back()}
      {...props}
    />
  );
};

export { BaseBackButton };
