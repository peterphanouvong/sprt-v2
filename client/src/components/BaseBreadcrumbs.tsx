import { ChevronRightIcon } from "@chakra-ui/icons";
import { Breadcrumb, BreadcrumbItem, BreadcrumbLink } from "@chakra-ui/react";
import React from "react";
import NextLink from "next/link";
import { Text } from "@chakra-ui/layout";

interface Props {
  crumbs: {
    title: string;
    href: string;
  }[];
}

const BaseBreadcrumbs: React.FC<Props> = ({ crumbs }) => {
  return (
    <Breadcrumb
      mb={4}
      spacing="8px"
      separator={<ChevronRightIcon color="gray.500" />}
    >
      {crumbs.map((crumb, index) => (
        <BreadcrumbItem key={index}>
          <BreadcrumbLink>
            <NextLink href={crumb.href}>
              <Text variant="body-3">{crumb.title}</Text>
            </NextLink>
          </BreadcrumbLink>
        </BreadcrumbItem>
      ))}
    </Breadcrumb>
  );
};

export { BaseBreadcrumbs };
