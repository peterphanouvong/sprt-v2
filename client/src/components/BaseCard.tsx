import { Box, BoxProps } from "@chakra-ui/react";
import React from "react";

type Props = BoxProps;

const BaseCard: React.FC<Props> = ({ ...props }) => {
  return (
    <Box
      borderRadius="sm"
      border="1px solid"
      bgColor="gray.50"
      borderColor="gray.200"
      padding={2}
      {...props}
    ></Box>
  );
};

export { BaseCard };
