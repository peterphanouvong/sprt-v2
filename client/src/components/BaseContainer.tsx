import { Box, BoxProps } from "@chakra-ui/react";
import React from "react";

type Props = BoxProps & {};

const BaseContainer: React.FC<Props> = ({ ...props }) => {
  return (
    <Box {...props} margin="auto" maxW="container.lg">
      {props.children}
    </Box>
  );
};

export { BaseContainer };
