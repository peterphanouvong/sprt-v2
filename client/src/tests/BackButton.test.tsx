import { shallow } from "enzyme";
import { BackButton } from "../components/BackButton";

test("BackButton Snapshot", () => {
  expect(shallow(<BackButton />)).toMatchSnapshot();
});
