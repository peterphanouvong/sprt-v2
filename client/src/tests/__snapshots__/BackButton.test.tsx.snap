// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Snapshot 1`] = `
<IconButton
  aria-label="back-button"
  colorScheme="gray"
  icon={<ChevronLeftIcon />}
  isRound={true}
  onClick={[Function]}
  variant="solid"
/>
`;
